components/centraldashboard-angular/frontend/src/app/services/namespace.service.ts

```diff
@@ -0,0 +1,199 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRoute, NavigationEnd, Params, Router } from '@angular/router';
+import { ReplaySubject } from 'rxjs';
+import { getQueryParams, getUrlFragment } from '../shared/utils';
+import { Namespace } from '../types/namespace';
+import { EnvironmentService } from './environment.service';
+import { LocalStorageService } from './local-storage.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CDBNamespaceService {
+  public namespacesSubject = new ReplaySubject<Namespace[]>();
+  public currentNamespaceSubject = new ReplaySubject<Namespace>();
+  public allNamespacesAllowedSubject = new ReplaySubject<boolean>();
+
+  private namespaces: Namespace[];
+  private user: string;
+  private allNamespacesAllowed: boolean;
+  private currentNamespace: Namespace | undefined;
+
+  readonly ALL_NAMESPACES_ALLOWED_LIST = [
+    'jupyter',
+    'volumes',
+    'tensorboards',
+    'katib',
+    'models',
+  ];
+  readonly ALL_NAMESPACES = 'All namespaces';
+  readonly allNamespacesOption: Namespace = {
+    namespace: this.ALL_NAMESPACES,
+    role: '',
+    user: '',
+  };
+
+  constructor(
+    private env: EnvironmentService,
+    private localStorage: LocalStorageService,
+    private router: Router,
+    private route: ActivatedRoute,
+  ) {
+    this.env.user.subscribe((user: string) => {
+      this.user = user;
+    });
+
+    this.env.namespaces.subscribe((namespaces: Namespace[]) => {
+      namespaces = [this.allNamespacesOption, ...namespaces];
+      this.namespaces = namespaces;
+      this.namespacesSubject.next(namespaces);
+
+      this.allNamespacesAllowed = this.calcAllNamespacesOption(this.router.url);
+      this.allNamespacesAllowedSubject.next(this.allNamespacesAllowed);
+
+      this.setCurrentNamespace(namespaces, this.user);
+    });
+
+    this.router.events.subscribe(event => {
+      if (event instanceof NavigationEnd) {
```

Comment:
We can use a `filter()` and remove the `if statement` since we only care for `NavigationEnd`